packages/core/src/customer/strategies/paypalcommerce/paypalcommerce-customer-strategy.ts

```diff
@@ -0,0 +1,411 @@
+import CustomerCredentials from '../../customer-credentials';
+import {
+    CustomerInitializeOptions,
+    CustomerRequestOptions,
+    ExecutePaymentMethodCheckoutOptions,
+} from '../../customer-request-options';
+import CustomerStrategy from '../customer-strategy';
+import PaypalCommerceCustomerInitializeOptions from './paypalcommerce-customer-initialize-options';
+
+export default class PaypalCommerceCustomerStrategy implements CustomerStrategy {
+    private _paypalCommerceSdk?: PaypalCommerceSDK;
+    private _onError = noop;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _customerActionCreator: CustomerActionCreator,
+        private _formPoster: FormPoster,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender,
+        private _consignmentActionCreator: ConsignmentActionCreator,
+        private _billingAddressActionCreator: BillingAddressActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+    ) {}
+
+    async initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { paypalcommerce, methodId } = options;
+
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.methodId" argument is not provided.',
+            );
+        }
+
+        if (!paypalcommerce) {
+            throw new InvalidArgumentError(
+                `Unable to initialize payment because "options.paypalcommerce" argument is not provided.`,
+            );
+        }
+
+        const { container, onError } = paypalcommerce;
+
+        this._onError = onError || noop;
+
+        if (!container) {
+            throw new InvalidArgumentError(
+                `Unable to initialize payment because "options.paypalcommerce.container" argument is not provided.`,
+            );
+        }
+
+        const state = await this._store.dispatch(
+            this._paymentMethodActionCreator.loadPaymentMethod(methodId),
+        );
+        const currencyCode = state.cart.getCartOrThrow().currency.code;
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+
+        this._paypalCommerceSdk = await this._paypalScriptLoader.getPayPalSDK(
+            paymentMethod,
+            currencyCode,
+        );
+
+        this._renderButton(methodId, paypalcommerce);
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    signIn(
+        credentials: CustomerCredentials,
+        options?: CustomerRequestOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(
```

Comment:
Just out of curiosity how will this be triggered? I ask since when we have paypal in checkout page wallet buttons will be at the top level? So we won't have the email address from the shopper 🤔

Reply:
As far as I remember, this method is repeating the default sigh in method without any effort for PayPal. The wallet buttons will be placed in customer step but in the top of the block, right?

I don't think that we should throw any error here. You can double check default customer strategy, also it has the same implementation.